### PR TITLE
Tests: assert !at_end in all get & next methods

### PR DESCRIPTION
**Pull request by Kijewski**
Most ranges will fail catastrophically if get() or next() was called when they are already at_end(). This PR makes it possible to find improper usages by adding assertions in all get() and next() methods.

### CODE REVIEW

File: include/rx/ranges.hpp [hunk 1]
```diff
@@ -605,17 +610,20 @@ struct idempotent_range {
...
     constexpr void next() noexcept {
+        RX_ASSERT(!at_end());
         inner_.next();
         if (!inner_.at_end()) {
             current_.emplace(inner_.get());
         }
     }
 
     constexpr bool at_end() const noexcept {
+        RX_ASSERT(!at_end());
```
Comment (Hazurl) [inline include/rx/ranges.hpp]: mistake ?

Comment (Kijewski) [reply include/rx/ranges.hpp]: Oops, thank you!

File: include/rx/ranges.hpp [hunk 2]
```diff
@@ -1475,6 +1500,7 @@ struct group_adjacent_by : private Compare {
         }
 
         void next() noexcept {
+            // No at_end() test: method is called in consructor.
```
Comment (Hazurl) [inline include/rx/ranges.hpp]: typo

Comment (Kijewski) [reply include/rx/ranges.hpp]: Thank you! Fixed.

**Review by Hazurl — APPROVED** [review]

**Review by simonask — APPROVED** [review]
Great! :+1:

**Comment by simonask** [comment]
Let me know when this is ready to merge.

**Comment by Kijewski** [comment]
I rebased it once more to let the MSVC tests run, but afterwards it is ready.

**Pull request merged**